apps/04TowersOfHanoi.js

```diff
@@ -13,30 +13,73 @@ var stacks = {
   c: []
 };
 
+
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
+var moveCount = 1; // store the total number of moves played by user;
+var legalMoveCount = 0;
+/* function movePiece will handle moving the user selected stacks from startStack to endStack */
+function movePiece(startStack, endStack) {
+  // Your code here : 
+   var temp; // store the popped stack value here, temp
 
-function movePiece() {
-  // Your code here
+    temp = stacks[startStack].pop(); // pop the user selected stack and store value in temp
+    stacks[endStack].push(temp); // push temp onto the user selected endstack
+   }
 
-}
+    // penguins are the best (wmy ife requested something about penguins)
 
-function isLegal() {
+/* isLegal will check to see if the user input stack push was legal*/
+function isLegal(startStack, endStack) {
   // Your code here
-
+  /* check if the the endStack is empty, or check if the stack being pushed is less than the
+     stack already on endStack */
+  var end = stacks[endStack].length; // store length of endStack in end;
+  var start = stacks[startStack].length; // store length of startStack in start;
+  if(stacks[endStack].length === 0 || stacks[startStack][start - 1] < stacks[endStack][end - 1]){
+    legalMoveCount++; // add 1 to legal move counter here
+    return true;
+  
+}
+  else{
+   // send invalid moves back
+    var temp = stacks[endStack].pop();
+    stacks[startStack].push(temp);
+    console.log("invalid move Try Again");
+
+return false;
+  }
 }
 
-function checkForWin() {
+function checkForWin(){
   // Your code here
-
+/* checks if the length of stacks 'b' or 'c' is 4, or, a completed game, provided all moves
+  were legal */
+if(stacks['b'].length === 4 || stacks['c'].length === 4){ 
```

Comment:
Lets make sure we have proper indention. Make sure everything within curly braces is indented over 2 spaces and that everything is aligned.